### PR TITLE
peer-812 keycat support(except tip question/answer)

### PR DESCRIPTION
**Pull request by s-kashko**

### CODE REVIEW

File: app/components/SignUpWrapper/SignUpOptions.jsx [hunk 1]
```diff
@@ -147,28 +148,34 @@ const RightMenuWithoutScatter = ({
...
-        processing={showScatterSignUpProcessing}
-        text={<FormattedMessage {...commonMessages.signUpViaWallet} />}
+    <Div>
+      <Footer
+        walletAction={showScatterSignUpForm}
+        signUpProcessing={
+          showScatterSignUpProcessing || showKeycatSignUpProcessing
+        }
+        keycatAction={showKeycatSignUpForm}
+        // keycatProcessing={
```
Comment (sergeyi23) [inline app/components/SignUpWrapper/SignUpOptions.jsx:163]: Is this commente code needed?

File: app/containers/SignUp/KeycatSignUpForm.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+import messages from './messages';
+
+import { Form } from './EmailVerificationForm';
+import loginMessages from '../Login/messages';
+import { REFERRAL_CODE } from '../Login/constants';
+import { getCookie } from '../../utils/cookie';
+import { REFERRAL_CODE_URI } from '../App/constants';
+
+const KeycatSignUpForm = ({ handleSubmit, keycatAccountValue, change }) => (
```
Comment (sergeyi23) [inline app/containers/SignUp/KeycatSignUpForm.jsx:36]: Most of this form can be moved into a component that can be reused for signup with Scatter and Sign with Keycat.

File: app/containers/SignUp/saga.js [hunk 3]
```diff
@@ -356,6 +364,49 @@ export function* signUpWithScatterWorker({ val }) {
   }
 }
 
+export function* signUpWithKeycatWorker({ val }) {
```
Comment (sergeyi23) [inline app/containers/SignUp/saga.js:367]: This code is duplication of `signUpWithScatterWorker`. Code has to be reused.

**Review by sergeyi23 — CHANGES_REQUESTED** [review]
There are some code duplications that need to be fixed

File: app/components/SignUpWrapper/SignUpOptions.jsx (hunk 1 above)
Comment (s-kashko) [reply app/components/SignUpWrapper/SignUpOptions.jsx]: fixed

File: app/containers/SignUp/KeycatSignUpForm.jsx (hunk 2 above)
Comment (s-kashko) [reply app/containers/SignUp/KeycatSignUpForm.jsx]: fixed

File: app/containers/SignUp/saga.js (hunk 3 above)
Comment (s-kashko) [reply app/containers/SignUp/saga.js]: fixed

File: app/utils/eosio.js [hunk 4]
```diff
@@ -104,6 +111,48 @@ class EosioService {
     this.selectedAccount = acc;
     this.withScatter = false;
     this.#key = key;
+
+    await this.initKeycat();
+    this.withKeycat = false;
```
Comment (sergeyi23) [inline app/utils/eosio.js:116]: These 2 lines can be moved to `initKeycat` function

File: app/containers/Login/actions.js [hunk 5]
```diff
@@ -42,24 +42,26 @@ export function showEmailPasswordForm(val) {
   };
 }
 
-// Login with scatter
+// Login with Wallet
 
-export function loginWithScatter() {
+export function loginWithWallet({ keycat, scatter }) {
```
Comment (sergeyi23) [inline app/containers/Login/actions.js:47]: optional: I see several places in the code where 2 boolean values are passed to indicate keycat vs scatter. Enums can simplify that logic and allow to store a single value. Also enum would not allow to send both values as true - keycat and scatter.

**Review by sergeyi23 — APPROVED** [review]

**Pull request merged**